core/cloudflow-flink/src/main/scala/cloudflow/flink/FlinkStreamlet.scala

```diff
@@ -167,12 +167,24 @@ abstract class FlinkStreamlet extends Streamlet[FlinkStreamletContext] with Seri
       StreamExecutionEnvironment.getExecutionEnvironment
     }
 
-    if (!env.getCheckpointConfig.isCheckpointingEnabled()) {
+    if (!env.getCheckpointConfig.isCheckpointingEnabled() && isDefaultCheckpointingEnabled(config, streamlet)) {
       setDefaultCheckpointing(env)
     }
     env
   }
 
+  /**
+   * This checks whether the user, through configuration, has disabled checkpointing
+   * by setting flink.execution.checkpointing.interval value to less then 0
+  **/
+  def isDefaultCheckpointingEnabled(config: Config, streamlet: String): Boolean = {
+    val runtimePath   = "cloudflow.runtimes.flink.config.cloudflow.checkpointing.default"
+    val streamletPath = s"cloudflow.streamlet.${streamlet}.config.cloudflow.checkpointing.default"
+    if (config.hasPath(streamletPath)) return config.getBoolean(streamletPath)
```

Comment:
Please 🙏  🙂 

```scala
if (config.hasPath(streamletPath)) {
  config.getBoolean(streamletPath)
} else if (config.hasPath(runtimePath)) {
  config.getBoolean(runtimePath)
} else {
  false
}

Reply:
of course! Sorry didn't read it the first time

Reply:
done now

Reply:
thanks!